### PR TITLE
Db week3

### PR DESCRIPTION
**Pull request by husenjanobul**

### CODE REVIEW

File: Week3/nodejs_tasks/routes/tasks.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        res.render('user/add', {
+            title: 'Add New User',
+            task_name: req.body.task_name
+        })
+    }
+})
+
+app.get('/edit/(:id)', (req, res, next) => {
+    req.getConnection((error, conn) => {
+        conn.query("SELECT * FROM tbl_tasks WHERE id = " +
```
Comment (mogaleaf) [inline Week3/nodejs_tasks/routes/tasks.js]: Be careful here you are not using prepared statement and string injection. You should not insert any values given by a call on your api even if it's your front end, you could vulnerable to attack.

File: Week3/nodejs_tasks/routes/tasks.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    req.assert('task_name', 'Name is required').noEmpty()
+
+    let errors = req.validationErrors();
+    if(!errors){
+        let task = {
+            task_name: req.sanitize('task_name').escape().trim()
+        }
+
+        req.getConnection((error, conn) => {
+            conn.query("UPDATE tbl_tasks SET ? WHERE id = " + req.params.id, task,
```
Comment (mogaleaf) [inline Week3/nodejs_tasks/routes/tasks.js]: Here you are mixing prepared statement and string injection, just use prepared statement.

File: Week3/nodejs_tasks/app.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+const dbOptions = {
+    host: config.databases.host,
+    user:config.databases.user,
+    password: config.databases.password,
+    port: config.databases.port,
+    databases: config.databases.db
+}
+app.use(myConnection(mysql, dbOptions, 'pool'));
+app.set('view engine', 'ejs');
+const index = require('./router/index');
```
Comment (mogaleaf) [inline Week3/nodejs_tasks/app.js]: here it should be routes not router...

**Comment by Tjebbee** [comment]
Hye @husenjanobul ; can you take a look at the personal feedaback and implement it ASAP.

**Pull request closed without merging**